### ISSUE TITLE
Type inference returns a type different from the declared type.

### ISSUE THREAD

**Issue by sasano8** [microsoft/pylance-release#1802]
In the following example,
passing `func3` will return `str` instead of the declared `Literal["running"]`.
It should return the declared type as is.

``` python
from typing import Literal, Tuple, List, Sequence, TypeVar, Iterable

T = TypeVar("T")

def func1(val: Iterable[T]) -> T:
    ...

def func2(val: Sequence[T]) -> T:
    ...

def func3(val: Tuple[T]) -> T:
    ...


status: Tuple[Literal["running"]] = ("running",)

a = func1(status)  # Literal["running"] to Literal["running"]
b = func2(status)  # Literal["running"] to Literal["running"]
c = func3(status)  # Literal["running"] to str why?
```

![Screenshot from 2021-09-11 12-12-53](https://user-images.githubusercontent.com/30338578/132936272-1ebbbf78-1050-4c2b-9035-87cb744adb48.png)

**Comment by erictraut** [comment]
Are you seeing a behavior that leads to a false positive or false negative type error, or are you just curious about the behavior you're seeing? If you are seeing an actual bug, please provide a sample that demonstrates the problem. If you're just curious, then posting a question in the discussion section would be more appropriate.

**Comment by sasano8** [comment]
I see this as a false positive type error.

I have only given it the type `Literal["running"]`.
So, TypeVar `T` is `Literal["running"]`.
And, `str` does not belong to `Literal["running"]`.
This is because `Literal["running"]` is narrower than str.

So it contradicts the next sentence.

>Return Type Inference
>The returned type is assumed to be the union of all types returned from all return statements.

The `str` that should not have belonged to `Literal["running"]` has returned.


I can't indicate the problem any further, and I'm not interested in it, so if it's not a problem, please close it.

kind regards.

**Comment by erictraut** [comment]
I agree that this is an inconsistency. I investigated and found the root cause. This inconsistency will be eliminated in the next release.

**Comment by sasano8** [comment]
Thank you!
I love VSCODE and I wish VSCODE further development.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.9.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202192-16-september-2021
